### PR TITLE
Storage lock fix

### PR DESCRIPTION
**Pull request by morganchen12**
Experimental fix (maybe) for #10463.

### CODE REVIEW

File: FirebaseStorage/Sources/Storage.swift [hunk 1]
```diff
@@ -302,6 +302,7 @@ import FirebaseAuthInterop
   }
 
   /// Map of apps to a dictionary of buckets to GTMSessionFetcherService.
+  private static let fetcherServiceLock = NSObject()
```
Comment (ncooke3) [inline FirebaseStorage/Sources/Storage.swift:305]: just a thought– in case this has similar behavior as an empty dictionary, maybe try creating a new UUID

Comment (morganchen12) [reply FirebaseStorage/Sources/Storage.swift:305]: `objc_sync_enter` uses the address so the object itself shouldn't matter. Dictionary being a value type with runtime optimizations messes with this according to [this blog post](https://straypixels.net/swift-dictionary-locking/), though the error we're running into isn't exactly the same.

Comment (paulb777) [reply FirebaseStorage/Sources/Storage.swift:305]: An NSObject should not have the same Swift runtime optimizations as a Swift dictionary.

This change fixes my repro example.

**Review by paulb777 — APPROVED** [review]
LGTM with a changelog.

The test failure is a common occasional flake

File: FirebaseStorage/Tests/Integration/StorageIntegration.swift [hunk 2]
```diff
@@ -137,6 +138,24 @@ class StorageResultTests: StorageIntegrationCommon {
     waitForExpectations()
   }
 
+  func testNoDeadlocks() throws {
```
Comment (paulb777) [inline FirebaseStorage/Tests/Integration/StorageIntegration.swift:141]: Nice!

File: FirebaseStorage/Tests/Integration/StorageIntegration.swift [hunk 3]
```diff
@@ -137,6 +138,24 @@ class StorageResultTests: StorageIntegrationCommon {
...
+    let storage2 = Storage.storage(url: "")
+
```
Comment (paulb777) [inline FirebaseStorage/Tests/Integration/StorageIntegration.swift:143]: Run style

**Pull request merged**